lib/data/landing_page_content_items/homepage.yaml

```diff
@@ -67,35 +67,59 @@ blocks:
 - type: grid_container
   blocks:
   - type: card
-    image:
-      alt: "Placeholder alt text"
-      source: "landing_page/placeholder/chart.png"
     card_content:
       blocks:
         - type: heading
           content: Korem ipsum dolor sit
-          path: http://gov.uk
+          path: /landing-page/task
           inverse: true
+          margin_bottom: 0
+        - type: statistics
+          title: "Chart to visually represent some data"
+          x_axis_label: "X Axis"
+          y_axis_label: "Y Axis"
+          csv_file: "data_one.csv"
+          data_source_link: https://www.ons.gov.uk/economy/grossdomesticproductgdp/timeseries/ihyq/qna
+          minimal: true
+          minimal_link: /landing-page/task
+          margin_bottom: 0
```

Comment:
It feels odd having this level of presentation data in what will ultimately become the content item.

Should this always be `0` if `minimal` is true? Could `minimal` be used to determine the `margin_bottom` instead?

Reply:
Yes, and that's much tidier, so I've made that change. I've also applied this to the `height` parameter, as that also makes sense.

However, I think this only works for this exact design, where we want to use these small charts in a three column layout - this wouldn't work for a more flexible system where the chart might appear elsewhere. I think at that point we could look at putting those overrides (for height, margin_bottom etc.) into the cards block, for that specific chart block.

Reply:
I've been having a think about this, as needing to set the `margin_bottom` affects the `heading` block too, and that doesn't have an `minimal` attribute that can be hijacked.

I think instead, the `card` view can be used to control how the blocks that appear inside it are rendered.

e.g.
Update `card` to pass an `options` hash to `render_block`

```ruby
<% block.card_content.each do |subblock| %>
    styling = { margin_bottom: 0 }

    <% if subblock.data["type"] == "heading" %>
      <div class="app-b-card__textbox">
        <%= render_block(subblock, styling:)  %>
      </div>
    <% elsif subblock.data["type"] == "statistics" %>
      <div class="app-b-card__chart">
        <%= render_block(subblock, styling:)  %>
      </div>
    <% else %>
      <%= render_block(subblock)  %>
    <% end %>
  <% end %>
```

Update the `render_block` method to accept an `options` hash:

```ruby
  def render_block(block, options: {})
    render("landing_page/blocks/#{block.type}", block:, options:)
  rescue ActionView::MissingTemplate
    Rails.logger.warn("Missing template for block #{block.type}")
    ""
  end
``` 

Lastly, update the views to read from the `options` hash

e.g. in the `statistics` partial add:

```ruby
<%
  add_gem_component_stylesheet("chart")

  margin_bottom = options[:margin_bottom] || 8
  height = block.data["height"]
%>
```

I wonder if something similar could be applied for the height?

The `heading` partial would have something similar

```ruby
<%
.
.
  margin_bottom = styling[:margin_bottom] || 6
%>
```

Reply:
@leenagupte great shout, have implemented this. It's spread a bit wider than this - turns out this is a great way for blocks to control `inverse` for subblocks.